### PR TITLE
TF-60 Suggestion email from contact device

### PR DESCRIPTION
**Pull request by dab246**
https://user-images.githubusercontent.com/80730648/134137828-6d1d2425-7db9-4ef7-a8e1-16ba3c22284f.mov

### CODE REVIEW

**Comment by chibenwa** [comment]
What happens if I refuse to give access to the device contact?

File: ios/Runner/Info.plist [hunk 1]
```diff
@@ -26,8 +26,10 @@
 	<string>LaunchScreen</string>
 	<key>UIMainStoryboardFile</key>
 	<string>Main</string>
-  <key>NSPhotoLibraryAddUsageDescription</key>
-  <string>To save photo to library, we need permission to access your photo library.</string>
+    <key>NSPhotoLibraryAddUsageDescription</key>
+    <string>To save photo to library, we need permission to access your photo library.</string>
+    <key>NSContactsUsageDescription</key>
+  	<string>You can choose to share files with these contacts.</string>
```
Comment (hoangdat) [inline ios/Runner/Info.plist:32]: `To send email with these contacts`

File: lib/features/composer/domain/usecases/get_autocomplete_with_device_contact_interactor.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/mail/email/email_address.dart';
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/composer/domain/model/auto_complete_pattern.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/get_autocomplete_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/state/get_device_contact_suggestions_state.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/get_autocomplete_interactor.dart';
+import 'package:tmail_ui_user/features/composer/domain/usecases/get_device_contact_suggestions_interactor.dart';
+
+class GetAutoCompleteWithDeviceContactInteractor {
```
Comment (hoangdat) [inline lib/features/composer/domain/usecases/get_autocomplete_with_device_contact_interactor.dart:12]: why not combine some contact sources to one interactor: `SearchEmailAddressInteractor`.

File: lib/features/composer/domain/repository/contact_repository.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+import 'package:model/model.dart';
+import 'package:tmail_ui_user/features/composer/domain/model/auto_complete_pattern.dart';
+
+abstract class ContactRepository {
+  Future<List<Contact>> getContactSuggestions(AutoCompletePattern autoCompletePattern);
```
Comment (hoangdat) [inline lib/features/composer/domain/repository/contact_repository.dart:6]: why not use `EmailAddress` to easy to merge with `getAutoComplete`

Comment (dab246) [reply lib/features/composer/domain/repository/contact_repository.dart:6]: I want it to be a common abstract for Contact so we can extend other contacts later. So I converted it to EmailAddress in interactor

**Comment by dab246** [comment]
> And I did not find the piece of code for Contact permission handler?. Please share the video for android device also.

I updated

https://user-images.githubusercontent.com/80730648/134275418-2ad021b9-3f0f-441e-9715-6c7ef115769d.mov

**Comment by chibenwa** [comment]
Hmmm I don't think we should prompt all the time the user to allow the user to access contacts if he keeps on refusing...

**Comment by dab246** [comment]
> Hmmm I don't think we should prompt all the time the user to allow the user to access contacts if he keeps on refusing...

On mobile platforms android & ios supports us to hide the dialog requesting permission when the user does not want it to be displayed with a button right on the dialog. Or it can check the number of rejections to hide it automatically.

**Comment by chibenwa** [comment]
No, keep it simple. It's fine if the plateform handles it natively.

**Comment by hoangdat** [comment]
> No, keep it simple. It's fine if the plateform handles it natively.

Please let me take responsible for the UX of the mobile. please only share your concern.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**